[docs/instrumenting] Add EMQ exporter

HagaiBarel opened the pull request:
This commit add [EMQ exporter](https://github.com/nuvo/emq_exporter) to the list of messaging systems exporters

Signed-off-by: Hagai Barel <hagai.barel@nuvo-group.com>

brian-brazil commented:
Thanks for sharing. This exporter is accepting passwords on the command line, which is not secure. Can you fix that?

I also noticed a few races. You should have scrape going directly to MustNewConstMetric, and remove e.metrics to fix those.

brian-brazil commented:
Whatever is not command line arguments. Files are best, but envvars are an option.

In Collect up is shared between scrapes, so might get things wrong with concurrent scrapes. Similarly with e.metrics.

HagaiBarel commented:
Alright, I'll work on both your points. 
Thanks!

brian-brazil commented:
Thanks!

HagaiBarel commented:
@brian-brazil Thanks for the merge!

Quick follow up to the points you raised - 
* I've deprecated the ability to pass credentials via command line, only via a file or `env var`
* I've introduced a `mutex` to handle concurrency better, refactored a lot of the code and added unit tests to detect potential data races

Thanks for the pointers, It really helped me to better rewrite things

brian-brazil commented:
> I've introduced a mutex to handle concurrency better, refactored a lot of the code and added unit tests to detect potential data races

When implemented in the normal way with ConstMetrics, you shouldn't need a mutex.
